### PR TITLE
Unit test category

### PR DESCRIPTION
**Pull request by zenithyap**
Added test cases for 
- useCategory.js
- Categories.js
- categoryModel.js

run `npm i` before running the test cases for categoryModel.js
 
For categoryModel.js, 2 of the test will not pass. I believe that the test case where name is present and slug is not present should be an error as well. Not too sure though, let me know what you guys think.

### CODE REVIEW

File: models/categoryModel.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        const validCategory = new Category(categoryData);
+        const savedCategory = await validCategory.save();
+
+        expect(savedCategory._id).toBeDefined();
+        expect(savedCategory.name).toBe(categoryData.name);
+        expect(savedCategory.slug).toBe(categoryData.slug);
+    });
+
+    // NEVER PASS
+    test('should throw error if name is missing', async () => {
```
Comment (eugenetangkj) [inline models/categoryModel.test.js:30]: Why should it throw an error if `name` is missing? I thought in `categoryModel.js`, both `name` and `slug` are not compulsory/required attributes?

**Review by eugenetangkj — COMMENTED** [review]
The tests seem great to me!

File: client/src/pages/Categories.test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const renderComponent = () => {
+    render(
+        <Router>
+            <Categories/>
+        </Router>
+    )
+};
+
+describe('Categories component', () => {
```
Comment (eugenetangkj) [inline client/src/pages/Categories.test.js:26]: Do you want to include a test case where `useCategory` returns empty array (i.e. No categories)

File: models/categoryModel.test.js (hunk 1 above)
Comment (zenithyap) [reply models/categoryModel.test.js:30]: For this I feel that it would be an error when name does not exists as you would just be adding an empty value into the database which does not make sense to me.

File: client/src/pages/Categories.test.js (hunk 2 above)
Comment (zenithyap) [reply client/src/pages/Categories.test.js:26]: Alright, I have added a test case for that

File: models/categoryModel.test.js (hunk 1 above)
Comment (eugenetangkj) [reply models/categoryModel.test.js:30]: However the schema did not state that these fields are `required`, so it would not throw an error?

image

Comment (eugenetangkj) [reply models/categoryModel.test.js:30]: Thoughts? @Aishwarya-Hariharan-Iyer @Kang-Quan @kohkaixun
